worker/src/request_init.rs

```diff
@@ -305,6 +329,98 @@ impl From<PolishConfig> for &str {
     }
 }
 
+/// Configuration options for Cloudflare's image resizing feature:
+/// <https://developers.cloudflare.com/images/image-resizing/>
+#[wasm_bindgen]
+#[derive(Clone, Default)]
+pub struct ResizeConfig {
+    pub anim: Option<bool>,
+    #[wasm_bindgen(skip)]
+    pub background: Option<String>,
+    #[wasm_bindgen(skip)]
+    pub blur: Option<String>,
+    pub brightness: Option<f64>,
+    pub contrast: Option<f64>,
+    pub dpr: Option<f64>,
+    pub fit: Option<ResizeFit>,
+    pub format: Option<ResizeFormat>,
+    pub gamma: Option<f64>,
+    // TODO
+    // #[wasm_bindgen(skip)]
+    // pub gravity: Option<ResizeGravity>,
+    pub height: Option<usize>,
+    pub metadata: Option<ResizeMetadata>,
+    pub onerror: Option<ResizeOnerror>,
+    pub quality: Option<usize>,
+    pub sharpen: Option<usize>,
+    pub trim: Option<ResizeTrim>,
+    pub width: Option<usize>,
+}
+
+#[wasm_bindgen]
+impl ResizeConfig {
+    #[wasm_bindgen(getter)]
+    pub fn background(&self) -> Option<String> {
+        self.background.clone()
+    }
+    #[wasm_bindgen(getter)]
+    pub fn blur(&self) -> Option<String> {
+        self.blur.clone()
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Copy)]
+pub enum ResizeFit {
+    ScaleDown,
```

Comment:
Values of the enum types (`ResizeFit`, `ResizeFormat`, `ResizeMetadata`, `ResizeOnerror`) will need to be converted with their target naming scheme (kebab scheme) accounted for, essentially like it's done above with `PolishConfig`.

For example, `ScaleDown` should be serialised into the string `scale-down`.

I see they other `&str` conversions are spelled out explicitly, which is a bit of a shame, since that means a lot of repetition. I would use [strum::IntoStaticStr](https://docs.rs/strum_macros/latest/strum_macros/derive.IntoStaticStr.html) but I don't know if the maintainers are okay with pulling in that dependency.